dev.Dockerfile

```diff
@@ -1,16 +1,30 @@
-FROM ruby:2.6.6
+#############
+## Stage 1 ##
+#############
+FROM ruby:2.6.6-alpine3.12 as gembuilder
 
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
-    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
-    curl -sL https://deb.nodesource.com/setup_14.x | bash - && \
-    apt-get install -y cmake
+RUN apk add --no-cache --update build-base cmake
+
+WORKDIR /usr/src/app
 
-RUN gem install bundler:2.1.4
 COPY Gemfile Gemfile.lock ./
-RUN bundle install
+RUN gem install bundler:2.1.4 && \
+    bundle install
+
+#############
+## Stage 2 ##
+#############
+FROM ruby:2.6.6-alpine3.12
 
 WORKDIR /usr/src/app
-ENTRYPOINT bundle update --full-index --conservative exercism_config && \
-           EXERCISM_DOCKER=true EXERCISM_ENV=development bundle exec setup_exercism_config && \
-           APP_ENV=development bundle exec rackup -p 3021 --host 0.0.0.0
+
+RUN gem install bundler:2.1.4
+COPY --from=gembuilder /usr/local/bundle/ /usr/local/bundle/
+
+# copy the source as late as possible to maximize cache
+COPY . .
+
+ENV CONTAINER_NAME=tooling-orchestrator
```

Comment:
What is this used for?

Reply:
Shell prompt, the container knowing "what am I", etc.

Reply:
Is this an official docker thing or something you've made up?

Reply:
Something I made up.

Reply:
So, this is here so that within the shell, you can do `echo $CONTAINER_NAME`?

Reply:
Yes or use it for any other purpose.  It seems like something that could be useful for container diag, etc.